charmcraft.yaml

```diff
@@ -23,7 +23,37 @@ bases:
           - amd64
 parts:
   charm:
+    charm-strict-dependencies: true
     charm-binary-python-packages:
+      - anyio
       - bcrypt
-    charm-python-packages: [setuptools, pip]
-    build-packages: [git]
+      - certifi
```

Comment:
Where did all these packages come from? I'm confused how you figured all it needed?

Reply:
With the latest release of charmcraft there were changes to dependency management.
To get this list of packages I `pip install -r requirements.txt` the existing requirements. That installs the direct and transitive dependencies and then I exported them to the requirements.txt file again. Previously this file only contained the direct dependencies.
Then I added them all to the charmcraft.yaml file to install them from pypi instead of building from source which often requires a C compiler or Rust compiler, etc.